sdk/include/reactos/shellutils.h

```diff
@@ -88,6 +88,42 @@ inline BOOL _ROS_FAILED_HELPER(HRESULT hr, const char* expr, const char* filenam
 } /* extern "C" */
 #endif /* defined(__cplusplus) */
 
+static inline UINT
+SHELL_ErrorBoxHelper(HWND hwndOwner, UINT Error)
+{
+    WCHAR buf[400];
+    UINT cch, msgId, u32_errstr = 2;
+
+    if (!IsWindowVisible(hwndOwner))
+        hwndOwner = NULL;
+    if (Error == ERROR_SUCCESS)
+        Error = ERROR_INTERNAL_ERROR;
+
+    msgId = Error;
+retry:
+    cch = FormatMessageW(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+                         NULL, msgId, 0, buf, _countof(buf), NULL);
+    if (!cch)
+    {
+        if (HIWORD(msgId) == HIWORD(HRESULT_FROM_WIN32(1)))
```

Comment:
What is the purpose of this part?

Reply:
If it is a HRESULT that was originally an ERROR_, it will try to find the message for ERROR_.

Reply:
Does this HIWORD stuff correspond to some of these macros? : https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/psdk/winerror.h;hb=1183e797b4c9e7fe97a78b7f3dfd2922a0598eff#l67

Reply:
No? I would have to use both HRESULT_FACILITY and HRESULT_SEVERITY to replace HIWORD.